jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -1794,29 +1794,33 @@ public boolean equalsIgnoreCase(String string) {
 			byte byteAtO1Last = helpers.getByteFromArrayByIndex(s1Value, s1len - 1);
 			byte byteAtO2Last = helpers.getByteFromArrayByIndex(s2Value, s1len - 1);
 
-			if (byteAtO1Last != byteAtO2Last &&
-					toUpperCase(helpers.byteToCharUnsigned(byteAtO1Last)) != toUpperCase(helpers.byteToCharUnsigned(byteAtO2Last))) {
+			if (byteAtO1Last != byteAtO2Last
```

Comment:
nitpick: please add parentheses for this condition, same for following similar conditions.